homeassistant/components/vesync/sensor.py

```diff
@@ -1,25 +1,125 @@
 """Support for power & energy sensors for VeSync outlets."""
+from collections.abc import Callable
+from dataclasses import dataclass
 import logging
 
 from homeassistant.components.sensor import (
     SensorDeviceClass,
     SensorEntity,
+    SensorEntityDescription,
     SensorStateClass,
 )
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import ENERGY_KILO_WATT_HOUR, POWER_WATT
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    ENERGY_KILO_WATT_HOUR,
+    PERCENTAGE,
+    POWER_WATT,
+)
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity import EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
 
-from .common import VeSyncBaseEntity
+from .common import VeSyncBaseEntity, VeSyncDevice
 from .const import DOMAIN, VS_DISCOVERY, VS_SENSORS
+from .fan import SKU_TO_BASE_DEVICE
```

Comment:
We should not depend on other platforms. Please move common constants to the `const.py` module.